File: FrontendWeb/src/declarations/dtos/output/OrderKwekerOutputDto.ts
// OrderKwekerOutputDto.ts
import {OrderStatus} from '../../enums/OrderStatus';
import {KoperInfoOutputDto} from './KoperInfoOutputDto';
import {OrderProductOutputDto} from './OrderProductOutputDto';

export interface OrderKwekerOutputDto {
	id: string;
	createdAt: string;
	status: OrderStatus;
	closedAt: string | null;
	quantity: number;
	totalPrice: number;
	products: OrderProductOutputDto[];
	koperInfo: KoperInfoOutputDto;
}

File: BackendAPI/Application/DTOs/Output/OrderKwekerOutputDto.cs
using Domain.Enums;

namespace Application.DTOs.Output;

public class OrderKwekerOutputDto
{
    public required Guid Id { get; set; }
    public required DateTimeOffset CreatedAt { get; set; }
    public required OrderStatus Status { get; set; }
    public required DateTimeOffset? ClosedAt { get; set; }
    public required int Quantity { get; set; }
    public required decimal TotalPrice { get; set; }
    public required List<OrderProductOutputDto> Products { get; set; }
    public required KoperInfoOutputDto KoperInfo { get; set; }
}
